### ISSUE TITLE
Feature Request: "Run once" or "This Day"

### ISSUE THREAD

**Issue by kolossboss** [nielsfaber/scheduler-card#34]
Great Card. Works and looks perfect. Thanks a lot!!

I would like to the ability to run a event just once.

For example:

I like to turn off my 3D printers after printing at 11:00am. But only one time. Not every day or every Monday.

I hope this is possible and its a cardtype compatible function.

Thanks a lot :-)

**Comment by nielsfaber** [comment]
Hi,
This functionality is asked before.
I think it is not difficult to implement.
The only problem i see is how to make it clear in the card, because it really conflicts with the ‘days’ option. But I could try to work out something and propose it.
Also I am not sure what the schedule should do after the event. Just remain idle until you set it again? Disable itself? Remove the schedule completely?
I’d like to hear your opinion about it.

**Comment by KTibow** [comment]
How about a repeat option, like repeat once, two times, choose a number, infinite?

**Comment by nielsfaber** [comment]
Could be, but would anyone use anything other then once or infinite? I wouldn’t know for which use-case...

**Comment by KTibow** [comment]
Fair point. As a work-around, you could also make a schedule to trigger on only the days that the schedule shouldn't trigger on to turn it off. The negative part is that you have 2 schedules to delete instead of just one...

**Comment by kolossboss** [comment]
Maybe you can call the button. „Single trigger“ or „trigger once“. And locate the button under custom.

**Comment by nielsfaber** [comment]
It will be added in a future release.
ETA for this feature will be begin October (hopefully sooner).
The function will not be part of the day selection, so you can program a rule for specific day(s) that will fire once.
The schedule will disable itself after firing (it will not be auto-removed from the list), you can enable it to run it again for 1 time.

**Comment by KTibow** [comment]
BTW if you wanted to have a use case for repeat between 1 and infinite:
- You're on vacation for a week. You only want to trigger the schedule for your vacation duration.
- You only want to run the sprinklers a second time a day during the hottest month of the summer.

**Comment by cibernox** [comment]
I opened a very similar issue in the integration repo yesterday. The idea was to have a way of setting a one time scheduler for the next hours. Very similar to how you can quickly put alarms for today or tomorrow in the clock apps in android.

This is slightly different and less generic than setting an alarm Once for any arbitrary time in the future, like in 10 days. I think that setting an scheduler to run once in a given date and hour is rather unusual (e.g. next Tuesday as 17:55), but setting a scheduler to run at the given hour (today the hour is in the future, tomorrow if the hour is in the past) less generic but pretty common.

I leave there some mocks of the interface to see if @kolossboss and @nielsfaber like the idea.

![alarm-for-tomorrow](https://user-images.githubusercontent.com/265339/94901281-b13b6f80-0496-11eb-892f-3c79fc0a004e.jpeg)
<img width="402" alt="scheduler-card-proposal" src="https://user-images.githubusercontent.com/265339/94901297-b993aa80-0496-11eb-9a75-11c3170acf24.png">
<img width="405" alt="scheduled-for-tomorrow-noon" src="https://user-images.githubusercontent.com/265339/94901304-bbf60480-0496-11eb-9f7f-8d7754fe673d.png">

**Comment by KTibow** [comment]
Also, inverting this idea, it'd be nice to be able to cancel a schedule, but just the next time it triggers. Maybe buttons could show up on hover, like Mail:
![image](https://user-images.githubusercontent.com/10727862/94929589-264c7c00-047a-11eb-9c31-e05ea9752b6d.png)
![image](https://user-images.githubusercontent.com/10727862/94929568-1cc31400-047a-11eb-8d13-e2c7cb7f10ee.png)

**Comment by nielsfaber** [comment]
The ‘run once’ option is added in release v1.7.5.
Try it out and see if you like it!

**Comment by KTibow** [comment]
Looks pretty good! Is there docs for it?

**Comment by kolossboss** [comment]
Thanks for adding this!! I created a Schedule with "Disable automatically after triggering" Works great so far.
But how can I enable this Schedule again? It would be great, if I can change the time and reenable it?!

**Comment by KTibow** [comment]
You can long-press to renable, and then single-click to edit and save.

**Comment by kolossboss** [comment]
> You can long-press to renable, and then single-click to edit and save.

Oh great. Thanks a lot :-)

**Comment by nielsfaber** [comment]
Closing because this functionality is done.
